tests/mock_crust/utils.rs

```diff
@@ -421,3 +448,67 @@ pub fn gen_bytes<R: Rng>(rng: &mut R, size: usize) -> Vec<u8> {
 pub fn gen_immutable_data<R: Rng>(rng: &mut R, size: usize) -> Data {
     Data::Immutable(ImmutableData::new(gen_bytes(rng, size)))
 }
+
+fn sanity_check(prefix_lengths: &[usize]) {
+    assert!(prefix_lengths.len() > 1,
+            "There should be at least two specified prefix lengths");
+    let sum = prefix_lengths.iter().fold(0, |accumulated, &bit_count| {
+        assert!(bit_count <= 8,
+                "The specified prefix lengths {:?} must each be no more than 8",
+                prefix_lengths);
+        accumulated + (1 << (8 - bit_count))
+    });
+    if sum < 256 {
+        panic!("The specified prefix lengths {:?} would not cover the entire address space",
+               prefix_lengths);
+    } else if sum > 256 {
+        panic!("The specified prefix lengths {:?} would require overlapping groups",
+               prefix_lengths);
+    }
+}
+
+fn prefixes<T: Rng>(prefix_lengths: &[usize], rng: &mut T) -> Vec<Prefix<XorName>> {
+    let mut prefixes = vec![Prefix::new(prefix_lengths[0], rng.gen())];
```

Comment:
Yes, I am sorting the prefixes before calling this function, but it would be safer to just do that inside too.  Or would you prefer I add a check to ensure the slice is sorted?